specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json

```diff
@@ -0,0 +1,140 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-01-01",
+    "title": "Schema of Azure IoT Hub events published to Azure Event Grid",
+    "description":
+      "Describes the schema of the Azure IoT Hub events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+  },
+  "paths": {},
+  "definitions": {
+    "IotHubDeviceCreatedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Devices.DeviceCreated event.",
+      "$ref" : "#/definitions/DeviceLifeCycleEventData"
```

Comment:
The right way to do this is:
```json
"IotHubDeviceCreatedEventData": {
    "properties":{
          "x-ms-client-flatten": true,
          "$ref": "#/definitions/DeviceLifeCycleEventData",
          "description": "device life cycle properties."
   }
}
```

Reply:
I would have to use the `"x-ms-client-flatten": true,` everywhere I reference is it?
```